dockerfiles/hpcc/templates/thor.yaml

```diff
@@ -1,84 +1,137 @@
 {{ range $thor := $.Values.thor -}}
 {{- $masterPort := $thor.masterport | default 20000 -}}
 {{- $slavePort := $thor.slaveport | default 20100 -}}
+{{- $eclagentName := printf "%s-agent" .name }}
 {{- $slaveName := printf "%s-slave" .name }}
 {{- $serviceName := printf "%s-svc" .name }}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ .name | quote }}
+  name: {{ $eclagentName | quote }}
 spec:
-  replicas: 1
+  replicas: {{ $thor.eclagent.replicas | default 1 }}
   selector:
     matchLabels:
-      run: {{ .name | quote }}
+      run: {{ $eclagentName | quote }}
   template:
     metadata:
       labels:
-        run: {{ .name | quote }}
+        run: {{ $eclagentName | quote }}
     spec:
-      {{- include "hpcc.checkDataStorageHostMount" (dict "root" $) | indent 6 }}
+      serviceAccountName: hpcc
       containers:
-      - name: {{ .name | quote }}
+      - name: {{ $eclagentName | quote }}
         ports:
           - containerPort: {{ $masterPort }}
         args: [
                 {{ include "hpcc.configArg" . }},
-                {{ include "hpcc.daliArg" $ }},
-                --masterport={{ $masterPort }},
-                --numSlaves={{ $thor.numSlaves }}
+                {{ include "hpcc.daliArg" $ }}
+                {{- if $thor.maxActive }},
+                --maxActive={{ $thor.maxActive }}
+                {{- end }}
               ]
 {{ include "hpcc.addSecurityContext" (dict "root" $ "me" .) | indent 8 }}
-{{ include "hpcc.addImageAttrs" (dict "root" $ "me" . "imagename" "thormaster") | indent 8 }}
+{{ include "hpcc.addImageAttrs" (dict "root" $ "me" . "imagename" "eclagent") | indent 8 }}
 {{ include "hpcc.addVolumeMounts" . | indent 8 }}
       volumes:
 {{ include "hpcc.addConfigVolume" . | indent 6 }}
 {{ include "hpcc.addVolumes" $ | indent 6 }}
 ---
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: {{ $slaveName | quote }}
-spec:
-  replicas: {{ $thor.numSlaves }}
-  selector:
-    matchLabels:
-      run: {{ $slaveName | quote }}
-  template:
+{{- include "hpcc.generateConfigMap" (dict "root" $ "component" "thor" "me" .) }}
+{{- if $thor.eclagent }}
+    eclagent:
+{{ toYaml $thor.eclagent | indent 6 }}
+      name: myeclagent-thor
```

Comment:
this looks too hardwired...

Reply:
should be there, was an experiment to work with matching environment.xml and differentiate between hthor and thor cluster as it is, but replaced by specifying queueNames.
Will remove.